elide-contrib/elide-dynamic-config-helpers/src/main/java/com/yahoo/elide/contrib/dynamicconfighelpers/validator/DynamicConfigValidator.java

```diff
@@ -118,59 +132,108 @@ public void readAndValidateConfigs() throws IOException {
     private void loadConfigMap() throws IOException {
         PathMatchingResourcePatternResolver resolver = new PathMatchingResourcePatternResolver(
                 this.getClass().getClassLoader());
-        Resource[] modelResources = resolver.getResources(this.configDir);
-        for (Resource resource : modelResources) {
-            this.resourceMap.put(resource.getFilename(), resource);
+        if (resolver.getResources(this.configDir).length == 0) {
+            throw new IllegalStateException(this.configDir + " : config path does not exist");
+        }
+        int configDirURILength = resolver.getResources(this.configDir)[0].getURI().toString().length();
+
+        Resource[] hjsonResources = resolver.getResources(this.configDir + HJSON_EXTN);
+        for (Resource resource : hjsonResources) {
+            this.resourceMap.put(resource.getURI().toString().substring(configDirURILength), resource);
         }
     }
 
     /**
      * Read variable file config.
-     * @return boolean true if variable config file exists else false
-     * @throws IOException
+     * @return Map<String, Object> A map containing all the variables if variable config exists else empty map
      */
-    private void readVariableConfig() throws IOException {
-        String key = Config.VARIABLE.getConfigPath();
-        if (this.resourceMap.containsKey(key)) {
-            String content = IOUtils.toString(this.resourceMap.get(key).getInputStream(), StandardCharsets.UTF_8);
-            this.setVariables(DynamicConfigHelpers.stringToVariablesPojo(content));
-            this.resourceMap.remove(Config.VARIABLE.getConfigPath());
-        }
+    private Map<String, Object> readVariableConfig(Config config) {
+
+        return this.resourceMap
+                        .entrySet()
+                        .stream()
+                        .filter(entry -> entry.getKey().startsWith(config.getConfigPath()))
```

Comment:
Should this be Config.VARIABLE.getConfigPath?

Reply:
No, Config.MODELVARIABLE / Config.DBVARIABLE are passed as parameter to this method.